### DISCUSSION TITLE
Resources from session 9: Chat Completion API Tools & Functions in RAG Chat Apps

### DISCUSSION THREAD

**Discussion by multispark** [microsoft/AI-Chat-App-Hack#73] [Announcements]
Demo Repo: http://aka.ms/ragchat
Video: https://www.youtube.com/watch?v=ccyF5nMiA68

**Comment by pamelafox** [comment]
Anthony's PR using tools for escalation: https://github.com/Azure-Samples/azure-search-openai-demo/pull/1176

**Reply by pamelafox** [reply]
OpenAI docs for function calling: https://platform.openai.com/docs/guides/function-calling

**Reply by pallavigadepalli** [reply]
I tried to follow along on the prompt but when I change the system settings to weather related prompt Anthony typed I get an out like this. The weather in Lisbon today is currently [insert current weather condition] with a temperature of [insert current temperature] degrees Celsius. Would you like more detailed information such as humidity, wind speed, and forecast for the day? what else do I need to change?

**Comment by pamelafox** [comment]
An example of using tools to alternatively search GitHub issue trackers: https://gist.github.com/pamelafox/a3fdea186b687509c02cb186ca203328

**Comment by JasonHaley** [comment]
the page he was showing on microsoft learn in first 10 min: [https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/function-calling?tabs=python](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/function-calling?tabs=python)
